composer/InstallMagoBinaryCommand.php

```diff
@@ -0,0 +1,268 @@
+            $executable_platform_content,
+        ): Response {
+            $phar = new \PharData($downloaded_file);
+            $phar->extractTo($release_dir);
+
+            $filesystem->remove($downloaded_file);
+
+            file_put_contents($executable_platform_file, $executable_platform_content);
+
+            return $response;
+        });
+
+        $io->write('');
+        $progress_bar = ($io instanceof ConsoleIO) ? $io->getProgressBar() : null;
+        $loop->wait([$promise], $progress_bar);
+        $io->write('');
+        $io->write('');
+        $io->write('Done!');
+
+        return self::SUCCESS;
+    }
+
+    private function detectMagoReleaseId(HttpDownloader $httpDownloader): string
+    {
+        $version = InstalledVersions::getVersion(MagoPlugin::PACKAGE_NAME);
+
+        $response = $httpDownloader->get($this->buildGithubApiUri('/releases?per_page=99999999999999999'));
+        $json = $response->decodeJson();
+
+        foreach ($json as $release) {
+            if ($release['tag_name'] === $version) {
+                return $release['id'];
+            }
+        }
+
+        return 'latest';
+    }
+
+    /**
+     * @return array{tag: string, downloads: array<string, {file: string, url: string}>}
+     */
+    private function buildAssetsMapForRelease(HttpDownloader $httpDownloader, string $releaseId): array
+    {
+        $response = $httpDownloader->get($this->buildGithubApiUri('/releases/' . $releaseId));
+        $json = $response->decodeJson();
+
+        return [
+            'tag' => $json['tag_name'],
+            'downloads' => array_reduce(
+                $json['assets'] ?? [],
+                /**
+                 * @param array<string, string> $downloadMap
+                 * @param array{browser_download_url: string, name: string} $asset
+                 * @return array<string, string>
+                 */
+                static function (array $downloadMap, array $asset): array {
+                    if (!str_ends_with($asset['name'], '.tar.gz') && !str_ends_with($asset['name'], '.zip')) {
+                        return $downloadMap;
+                    }
+
+                    $platform = preg_replace('{^(.*)(\.tar\.gz|\.zip)$}', '$1', $asset['name']);
+
+                    $downloadMap[$platform] = [
+                        'file' => $asset['name'],
+                        'url' => $asset['browser_download_url'],
+                    ];
+
+                    return $downloadMap;
+                },
+                [],
+            ),
+        ];
+    }
+
+    private function buildGithubApiUri(string $path): string
+    {
+        return 'https://api.github.com/repos/carthage-software/mago' . $path;
+    }
+
+    /**
+     * @return array{platform: string, executable: string, storage_dir: string}
+     */
+    private function detectPlatformInfo(ProcessExecutor $process_executor, string $version): array
+    {
+        $arch_name = php_uname('m');
```

Comment:
`php_uname('m')` will return an uppercase string on windows so you should lowercase `$arch_name` like you did with `$os` later :+1: